sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -184,19 +184,16 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
     case RefreshTable(r: ResolvedTable) =>
       RefreshTableExec(r.catalog, r.identifier, recacheTable(r)) :: Nil
 
-    case ReplaceTable(catalog, ident, schema, parts, props, orCreate) =>
-      val newProps = props.get(TableCatalog.PROP_LOCATION).map { loc =>
-        props + (TableCatalog.PROP_LOCATION -> makeQualifiedDBObjectPath(loc))
-      }.getOrElse(props)
-      val propsWithOwner = CatalogV2Util.withDefaultOwnership(newProps)
+    case ReplaceTable(ResolvedDBObjectName(catalog, ident), schema, parts, tableSpec, orCreate) =>
+      val qualifiedLocation = tableSpec.location.map(makeQualifiedDBObjectPath(_))
```

Comment:
not related to this PR: seems path is not qualified in `CreateTableAsSelect`? cc @imback82

Reply:
Right, it doesn't seem to be qualified. I will create a PR for that.